standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/service/AcidHouseKeeperService.java

```diff
@@ -38,16 +40,22 @@ public class AcidHouseKeeperService implements MetastoreTaskThread {
   private static final Logger LOG = LoggerFactory.getLogger(AcidHouseKeeperService.class);
 
   private Configuration conf;
-  private boolean isCompactorEnabled;
   private TxnStore txnHandler;
+  private String serviceName;
+
+  public void setServiceName(String serviceName) {
```

Comment:
why do we need to expose `setServiceName` & `getTxnHandler`?

Reply:
To get the service name printed in log line in run method as we are not overriding it in CompactionHouseKeeper.java
   ```
- LOG.info("Starting to run AcidHouseKeeperService.");
   + LOG.info("Starting to run {}", serviceName);
```

And reuse TxnHandler created already in AcidHouseKeeperService as we are overriding cleanTheHouse method

 ```
void cleanTheHouse() {
    if (isCompactorEnabled) {
      performTask(getTxnHandler()::removeDuplicateCompletedTxnComponents,
          "Cleaning duplicate COMPLETED_TXN_COMPONENTS entries");
      performTask(getTxnHandler()::purgeCompactionHistory, "Cleaning obsolete compaction history entries");
    }
  }
```

Please guide me if CompactionHouseKeeper needs to be completely decoupled with AcidHouseKeeperService code wise by not implementing inheritance

Reply:
yes, it should be an independent MetastoreTaskThread service